contracts/pool/src/contract.rs

```diff
@@ -408,12 +362,25 @@ impl LiquidityPoolTrait for LiquidityPool {
         let total_shares = utils::get_total_shares(&env);
 
         let new_total_shares = if pool_balance_a > 0 && pool_balance_b > 0 {
-            let shares_a = (balance_a * total_shares) / pool_balance_a;
-            let shares_b = (balance_b * total_shares) / pool_balance_b;
-            shares_a.min(shares_b)
+            // use 10_000 multiplier to acheieve a bit bigger precision
+            let shares_a = (balance_a * total_shares) * 10_000 / pool_balance_a;
+            let shares_b = (balance_b * total_shares) * 10_000 / pool_balance_b;
+            shares_a.min(shares_b) / 10_000
```

Comment:
Those are not needed anymore I think.

```suggestion
            let shares_a = (balance_a * total_shares) / pool_balance_a;
            let shares_b = (balance_b * total_shares) / pool_balance_b;
            shares_a.min(shares_b)
```